### PR TITLE
OHFJIRA-51: Cluster node management API.

### PR DESCRIPTION
**Pull request by hanusto**
Exposed endpoint for management cluster node entities via REST calls.

API design: http://docs.openhubframework.apiary.io/#reference/0/openhub-cluster-nodes-management
Mockup design: https://openwise.mybalsamiq.com/projects/openhub/Cluster%20nodes

## Changes

- fixed javadoc for `NodeService` API
- added implementation via `ClusterNodeController`
- added test to verify functionality
- added `CollectionWrapper` class to hold RPC items

Issue: OHFJIRA-51

### CODE REVIEW

File: web-admin/src/main/java/org/openhubframework/openhub/admin/web/cluster/node/rpc/NodeRpc.java [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    }
+
+    /**
+     * Validates RPC (mandatory attributes, correct object state etc.).
+     *
+     * @param errors       Binding/validation errors
+     * @param updateEntity {@code null} if new entity is created otherwise entity that is updated
+     * @throws ValidationException when there is error in input data
+     */
+    public final void preSaveValidation(BindingResult errors, @Nullable Node updateEntity) throws ValidationException {
```
Comment (pjuza) [inline web-admin/src/main/java/org/openhubframework/openhub/admin/web/cluster/node/rpc/NodeRpc.java:182]: Why this method when you can call directly validate() method?

File: web-admin/src/main/java/org/openhubframework/openhub/admin/web/cluster/node/rest/ClusterNodeController.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        Assert.notNull(nodeRpc, "nodeRpc can not be null");
+        if (nodeRpc.getId() != null) {
+            Constraints.state(id.equals(nodeRpc.getId()), "IDs must be equal");
+        }
+
+        // get entity to update it
+        Node dbNode = nodeService.getNodeById(id);
+
+        // save entity
+        nodeService.update(dbNode, new NodeRpc.ChangeCallback(errors, nodeRpc));
```
Comment (pjuza) [inline web-admin/src/main/java/org/openhubframework/openhub/admin/web/cluster/node/rest/ClusterNodeController.java:104]: I see that validation concept doesn't go well with concept of ChangeCallback ... It's statement, I would leave it and we will see ...

File: web-admin/src/main/java/org/openhubframework/openhub/admin/web/common/rpc/CollectionWrapper.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+
+/**
+ * Collection wrapper of items.
+ *
+ * @param <T> - type of item in collection result
+ * @author Tomas Hanus
+ * @since 2.0
+ */
+public class CollectionWrapper<T> {
```
Comment (pjuza) [inline web-admin/src/main/java/org/openhubframework/openhub/admin/web/common/rpc/CollectionWrapper.java:35]: I like it ... 👍

**Review by pjuza — APPROVED** [review]

File: web-admin/src/main/java/org/openhubframework/openhub/admin/web/cluster/node/rpc/NodeRpc.java (hunk 1 above)
Comment (hanusto) [reply web-admin/src/main/java/org/openhubframework/openhub/admin/web/cluster/node/rpc/NodeRpc.java:182]: Cannot. Validation is protected and not visible from outside.

File: web-admin/src/main/java/org/openhubframework/openhub/admin/web/cluster/node/rest/ClusterNodeController.java (hunk 2 above)
Comment (hanusto) [reply web-admin/src/main/java/org/openhubframework/openhub/admin/web/cluster/node/rest/ClusterNodeController.java:104]: Yes, but I think this is exception :).

File: web-admin/src/main/java/org/openhubframework/openhub/admin/web/cluster/node/rest/ClusterNodeController.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+    /**
+     * Updates existing {@link NodeRpc}.
+     *
+     * @param id      as identifier of {@link NodeRpc}
+     * @param nodeRpc The {@link NodeRpc}
+     * @throws ValidationException  if input object has wrong values
+     * @throws NoDataFoundException entity not found for update
+     */
+    @RequestMapping(value = "/{id}", method = RequestMethod.PUT, produces = {"application/xml", "application/json"})
```
Comment (kkovarik) [inline web-admin/src/main/java/org/openhubframework/openhub/admin/web/cluster/node/rest/ClusterNodeController.java:89]: Philosophical: Is the **produces** configuration needed with 204 No Content response? As API consumer won´t be expecting any kind of response, he may not expect to send any Accept* headers?

File: web-admin/src/main/java/org/openhubframework/openhub/admin/web/common/rpc/CollectionWrapper.java (hunk 3 above)
Comment (kkovarik) [reply web-admin/src/main/java/org/openhubframework/openhub/admin/web/common/rpc/CollectionWrapper.java:35]: +1

**Review by kkovarik — APPROVED** [review]

**Pull request merged**